vadl/test/vadl/lcb/riscv/riscv32/GenerateLinkerComponentsRiscv32PassTest.java

```diff
@@ -0,0 +1,263 @@
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import vadl.gcb.passes.relocation.model.CompilerRelocation;
+import vadl.gcb.passes.relocation.model.Modifier;
+import vadl.gcb.valuetypes.RelocationFunctionLabel;
+import vadl.gcb.valuetypes.VariantKind;
+import vadl.lcb.AbstractLcbTest;
+import vadl.lcb.passes.relocation.GenerateLinkerComponentsPass;
+import vadl.pass.PassKey;
+import vadl.pass.exception.DuplicatedPassKeyException;
+
+public class GenerateLinkerComponentsRiscv32PassTest extends AbstractLcbTest {
+
+  @Test
+  void shouldGenerateExpectedModifiers() throws DuplicatedPassKeyException, IOException {
+    var setup = runLcb(getConfiguration(false), "sys/risc-v/rv32im.vadl",
+        new PassKey(GenerateLinkerComponentsPass.class.getName()));
+    var passManager = setup.passManager();
+
+    var generatedLinkerComponents =
+        (GenerateLinkerComponentsPass.Output) passManager.getPassResults()
+            .lastResultOf(GenerateLinkerComponentsPass.class);
+
+    var generatedModifiers = generatedLinkerComponents.modifiers();
+    expectedModifiers().forEach(
+        expected -> Assertions.assertTrue(
+            generatedModifiers.stream()
+                .anyMatch(generated -> generated.value().equals(expected.value())
+                    && generated.kind().equals(expected.kind())
+                    &&
+                    generated.relocationFunctionLabel().equals(expected.relocationFunctionLabel())
+                ),
+            "Expected modifier: " + expected + " not found in generated modifiers: "
+                + generatedModifiers
+        )
+    );
+  }
+
+  private static Stream<Modifier> expectedModifiers() {
+    CompilerRelocation.Kind ABS = CompilerRelocation.Kind.ABSOLUTE;
+    CompilerRelocation.Kind REL = CompilerRelocation.Kind.RELATIVE;
+
+    return Stream.of(
+        new Modifier("MO_RV3264I_hi", ABS, Optional.of(RelocationFunctionLabel.HI)),
+        new Modifier("MO_RV3264I_lo", ABS, Optional.of(RelocationFunctionLabel.LO)),
+        new Modifier("MO_ABS_RV3264I_Itype_imm", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Itype_imm", REL, Optional.empty()),
+        new Modifier("MO_ABS_RV3264I_Utype_imm", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Utype_imm", REL, Optional.empty()),
+        new Modifier("MO_ABS_RV3264I_Stype_imm", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Stype_imm", REL, Optional.empty()),
+        new Modifier("MO_ABS_RV3264I_Btype_imm", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Btype_imm", REL, Optional.empty()),
+        new Modifier("MO_ABS_RV3264I_Jtype_imm", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Jtype_imm", REL, Optional.empty()),
+        new Modifier("MO_ABS_RV3264I_Ftype_sft", ABS, Optional.empty()),
+        new Modifier("MO_REL_RV3264I_Ftype_sft", REL, Optional.empty())
```

Comment:
minor: please also add the tests for `pcrel_hi`, `pcrel_lo` and `got_hi`

`to32AndHi`, `to32AndLo`, `lowerHalfHi` and `lowerHalfLo` can be skipped